src/docs/truffle/reference/configuration.md

```diff
@@ -166,27 +166,123 @@ mocha: {
 }
 ```
 
-## Solidity compiler configuration
+## Compiler configuration
 
-Solidity compiler settings. Supports optimizer settings for `solc`.
+In the `compilers` object you can specify settings related to the compilers used by Truffle.
 
 ### solc
 
-Configuration options to pass to the Solidity compiler.
+Solidity compiler settings. Supports optimizer settings for `solc`.
 
-Example:
+You may specify...
++ any solc-js version listed at [solc-bin](http://solc-bin.ethereum.org/bin/list.json). Specify the one you want and Truffle will get it for you.
++ a natively compiled solc binary (you'll need to install this yourself, links to help below).
++ a dockerized solc from one of images published [here](https://hub.docker.com/r/ethereum/solc/tags/). (You'll also need to pull down the docker image yourself but it's really easy.)
++ a path to a locally available solc
+
+Truffle config example:
 
 ```javascript
-solc: {
-  optimizer: {
-    enabled: true,
-    runs: 200
+module.exports = {
+  compilers: {
+    solc: {
+      version: <string>,     // A version or constraint - Ex. "^0.5.0"
+      optimizer: {
+        enabled: <boolean>,
+        runs: <number>       // Optimize for how many times you intend to run the code
+      }
+    }
   }
 }
 ```
 
 For more information, please see the Solidity documentation on [Compiler Input and Output JSON Description](http://solidity.readthedocs.io/en/develop/using-the-compiler.html#compiler-input-and-output-json-description).
 
+### external compilers
+
+For more advanced use cases, let's say your smart contract development workflow involves more than just compiling Solidity contracts. Maybe you're writing eWASM precompiles or making a two-dimensional delegatecall proxy. Or maybe you would just rather use @pubkey's solidity-cli instead of Truffle's solc configuration.
```

Comment:
This paragraph needs links

Reply:
I lifted this out of the release notes because it seemed pretty thorough, I'll go ahead and keep the same links as in the notes.

Reply:
@gnidan Do you think we should keep all those references in the docs or should we make it more general for the docs?

Reply:
More general would be my guess. If we want to include those references, we might just make a section "Example Projects".

WDYT?

Reply:
Maybe we should just leave this more abstract.
`For more advanced use cases with artifact creation you can use the external compilers configuration.`

Reply:
Then it gives some examples of different use cases below.

Reply:
Yep sounds good to me!